crates/polars-plan/src/plans/optimizer/simplify_functions.rs

```diff
@@ -7,6 +7,58 @@ pub(super) fn optimize_functions(
     expr_arena: &mut Arena<AExpr>,
 ) -> PolarsResult<Option<AExpr>> {
     let out = match function {
+        // is_null().any() -> null_count() > 0
+        // is_not_null().any() ->  null_count() < len()
+        FunctionExpr::Boolean(BooleanFunction::Any { ignore_nulls: _ }) => {
+            let input_node = expr_arena.get(input[0].node());
+            match input_node {
+                AExpr::Function {
+                    input,
+                    function: FunctionExpr::Boolean(BooleanFunction::IsNull),
+                    options: _,
+                } => Some(AExpr::BinaryExpr {
+                    left: expr_arena.add(make_null_count_expr!(input)),
+                    op: Operator::Gt,
+                    right: expr_arena.add(AExpr::Literal(LiteralValue::Int8(0))),
```

Comment:
You should use a literal type based on `IdxSize` - you would need to do something like -

```rust
if std::mem::size_of::<IdxSize>() == 8 {
    LiteralValue::UInt64
} else {
    LiteralValue::UInt32
}
```

I think you need this a few times so you can probably add this as a function under `impl LiteralValue` -
```rust
fn new_idxsize(value: IdxSize) -> Self {
    ...
}
```

Reply:
Are you sure I can add this? `LiteralValue` is enum, not a struct.

Maybe I should add macro rule for IdxSize to implement the `Literal` trait?

Reply:
Clippy made me some problems (in `make pre-commit`) using this approach, although I added `#[allow(clippy::useless_conversion)]`.

I changed to new trait, please tell me what you think, thanks.

Reply:
We shouldn't make a trait for this - it introduces another way to create a LiteralValue that isn't really discoverable. We should just create the function under `impl LiteralValue` to be consistent. Can you post the clippy error that you get?

Reply:
Sure :)

My original function was:

```rust
pub(crate) fn new_idxsize(value: IdxSize) -> Self {
        if size_of::<IdxSize>() == 8 {
            let v = value as u64;
            LiteralValue::UInt64(value as u64)
        } else {
            LiteralValue::UInt32(value as u32)
        }
    }
```
And I got the clippy error:
`error: casting to the same type is unnecessary (`u32` -> `u32`)`

I managed to workaround now with:
```rust
pub(crate) fn new_idxsize(value: IdxSize) -> Self {
        if size_of::<IdxSize>() == 8 {
            #[allow(clippy::unnecessary_cast)]
            let value = value as u64;
            LiteralValue::UInt64(value)
        } else {
            #[allow(clippy::unnecessary_cast)]
            let value = value as u32;
            LiteralValue::UInt32(value)
        }
    }
```

What do ypu think?

Reply:
Ah, I can see the problem - to be 100% correct with clippy here we should use the feature gates - can you try this?
```rust
pub(crate) fn new_idxsize(value: IdxSize) -> Self {
    #[cfg(feature = "bigidx")]
    {
        LiteralValue::UInt64(value)
    }
    #[cfg(not(feature = "bigidx"))]
    {
        LiteralValue::UInt32(value)
    }
}
```

Reply:
Works great! thanks, pushed a fix.